### PR TITLE
[it] Add integration test against simple.temple

### PR DESCRIPTION
**Pull request by jaylees14**
Turns out `simple.temple` doesn't compile 😫... so let's make it an integration test.
I couldn't be bothered to build a `Templefile` that matches `simple.temple` so I'm using the normal pipeline of raw string → `Templefile`. This means if you make change to `simple.temple`, it should now break the integration tests.

Here are the current errors - I've commented the actual test out so it's easier to see what the fixes were: 

Screenshot 2020-04-07 at 19 01 00

### CODE REVIEW

File: src/it/scala/temple/generate/server/GoGeneratorIntegrationTest.scala [hunk 1]
```diff
@@ -85,4 +91,26 @@ class GoGeneratorIntegrationTest extends GolangSpec with Matchers with BeforeAnd
...
+
+    analyzedTemplefile.allServices.keys.map(StringUtils.kebabCase).foreach { service =>
+      val filesInService = project.files.filter { case (file, _) => file.folder.startsWith(service) }
+      val validationErrors = validateAll(
+        filesInService,
+        entryFile = File(service, s"$service.go"),
+      )
+
+    // TODO: Uncomment me, and fix the bugs!
+//      validationErrors shouldBe empty
```
Comment (c-harding) [inline src/it/scala/temple/generate/server/GoGeneratorIntegrationTest.scala]: ooh, exciting

**Review by c-harding — APPROVED** [review]
nice test, 100% pass rate

File: src/it/scala/temple/generate/server/GoGeneratorIntegrationTest.scala (hunk 1 above)
Comment (willjvsmith) [reply src/it/scala/temple/generate/server/GoGeneratorIntegrationTest.scala]: dealing with it now

**Review by willjvsmith — APPROVED** [review]

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainHandlersGenerator.scala [hunk 2]
```diff
@@ -19,12 +19,16 @@ import scala.Option.when
...
 object GoServiceMainHandlersGenerator {
 
-  private def generateResponseMapFormat(attributeType: AttributeType): String =
-    // Must add formatting to attributes with datetime type
-    attributeType match {
-      case DateTimeType => s".${genFunctionCall("Format", "time.RFC3339")}"
+  private def generateResponseMapFormat(attributeType: AttributeType): String = {
+    // Must add formatting to attributes with date, time or datetime type
+    val layout = attributeType match {
+      case DateType     => doubleQuote("2020-12-31")
```
Comment (jaylees14) [inline src/main/scala/temple/generate/server/go/service/main/GoServiceMainHandlersGenerator.scala]: @smithwjv See my gist (https://gist.github.com/jaylees14/e86b262ca4e0b8185cb5357c78fc01f3) for the correct format here. You need to use the reference time: https://gobyexample.com/time-formatting-parsing

**Review by Lewiky — APPROVED** [review]
LGTM

File: src/main/scala/temple/generate/server/go/service/main/GoServiceMainHandlersGenerator.scala (hunk 2 above)
Comment (willjvsmith) [reply src/main/scala/temple/generate/server/go/service/main/GoServiceMainHandlersGenerator.scala]: Was wondering how they told the difference between days and months aha

**Pull request merged**